routes/folders.js

```diff
@@ -0,0 +1,36 @@
+const express = require('express');
+const router = express.Router();
+
+// Import middleware
+const { attachRouteHandlerWrapper } = require('../middleware/routeHandler')
+
+// Import classes 
+const { File, CollectionPageType } = require('../classes/File.js');
+
+// Import error
+const { BadRequestError } = require('../errors/BadRequestError')
+
+// List pages and directories in folder
+async function listFolderContent (req, res, next) {
+    const { accessToken } = req
+    const { siteName } = req.params
+    const { path } = req.query // paths begin with _
+
+    // Validation of path
+    const pathArr = path.split('/')
+    if (path.slice(0,1) !== '_' || pathArr.length > 2 || (pathArr.length === 2 && pathArr[1].includes('.'))) {
```

Comment:
This validation maybe can be left to a "higher level"

Reply:
Validation has been shifted to the `list` method for the `Directory` class so that it is performed only conditionally (when the `dirType` is `FolderType`) - see commit [3619f17](https://github.com/isomerpages/isomercms-backend/pull/108/commits/3619f174823f9754dc16335ec33a823521e6fb8b)